packages/message/src/types.ts

```diff
@@ -1,15 +1,29 @@
-export type MessageTemplate = {
-  cumulus_meta: object
+import { Message } from '@cumulus/types';
+
+export interface MessageTemplateCumulusMeta {
+  queueExecutionLimits: {
```

Comment:
Also, can you explain the distinction between this and `Message.CumulusMeta`?

Reply:
See the answer below for some initial context on types in this package vs `types`. 

But fundamentally, this type is only trying to specify the properties that are added to `cumulus_meta` via the message template, which is `queueExecutionLimits` comes from , so the typing here reflects that in a nice way. And I think wanted to be strict about the fact that `cumulus_meta.queueExecutionLimits` should always exist on a queued message rather than making it optional (`?`) on the generic `Message.CumulusMeta` type. 

Doing it this way allows me to access it confidently on a queued message a non-optional property, like so:

https://github.com/nasa/cumulus/pull/1782/files?file-filters%5B%5D=.ts#diff-a3eda81810c6954b0973039cd9ec28a2R87

But truthfully, I think there was some trying of different options as I implemented this and this felt like the best option that also made the TS compiler happy

Reply:
Makes sense.